### PR TITLE
Integration Test Fixes

### PR DESCRIPTION
**Pull request by nholland94**
This PR contains a variety of fixes to the integration test framework to address recent issues. Fixes included:

- graphql ingress race condition (nodes failing to initialize, but initialization log is present)
- context deadline timeout on network destructions
- skip deploying graphql ingresses for tests that do not require it
- pod assignment race condition (very rare, but became more frequent with changes introduced in this PR)
- seed node artifact generation race condition (nodes failing to initialize, unable to find initial peers and restart in time)

### CODE REVIEW

File: dockerfiles/Dockerfile-coda-daemon-puppeteered
Comment (QuiteStochastic) [inline dockerfiles/Dockerfile-coda-daemon-puppeteered:26]: wasn't there a reason we were grabbing the py file from github rather than simply ADD-ing it into the container?  although i suppose if this works, then it works

File: automation/terraform/testnets/it-natha-8c57f100-bootstrap/main.tf.json
Comment (QuiteStochastic) [inline automation/terraform/testnets/it-natha-8c57f100-bootstrap/main.tf.json:1]: you're going to remove this file before merge right?

File: dockerfiles/Dockerfile-coda-daemon-puppeteered
Comment (nholland94) [reply dockerfiles/Dockerfile-coda-daemon-puppeteered:26]: Yes, this was checked in by accident.

File: src/app/test_executive/test_executive.ml
Comment (QuiteStochastic) [inline src/app/test_executive/test_executive.ml:306]: now that we do this, is it still necessary to wait_for_init at the beginning of every integration test?

**Review by QuiteStochastic — APPROVED** [review]

File: src/app/test_executive/test_executive.ml
Comment (nholland94) [reply src/app/test_executive/test_executive.ml:306]: Yes, this does not wait for init, it just starts the nodes.

**Comment by aneesharaines** [comment]
!approved-for-mainnet

**Review by lk86 — APPROVED** [review]

**Pull request merged**
